### PR TITLE
Document burrow deploy and wasm

### PR DESCRIPTION
**Pull request by seanyoung**
Signed-off-by: Sean Young <sean.young@monax.io>

### CODE REVIEW

File: docs/reference/deploy-yaml.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+* create proposals or vote for a proposal
+
+burrow deploy needs a script to its commands. This script format bares some similarity to [ansible](https://www.ansible.com/). It
+is in yaml format. The top level structure is an array of [jobs](https://github.com/hyperledger/burrow/blob/develop/deploy/def/job.go).
+The different job types are [defined here](https://github.com/hyperledger/burrow/blob/develop/deploy/def/jobs.go).
+
+Each job has a name. This name can be used in later jobs to refer to the result of a previous job (e.g. the address of a contract
+which was deployed). The jobs are executed in-order.
+
+Whereever an account is specified, this can also be accessed via the burrow keys keyname.
```
Comment (gregdhill) [inline docs/reference/deploy-yaml.md]: typo

File: docs/reference/deploy-yaml.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+* _data_: the arguments to the contract's constructor
+
+The solidity source file is compiled using the [solidity compiler](https://github.com/ethereum/solidity) unless the `--wasm` argument was given
+on the burrow deploy command line, in which case the [solang compiler](https://github.com/ethereum/solidity) is used.
+
+The contract is deployed with its metadata, so that we can retrieve the ABI when we need to call a function of this contract. For this
+reason, the bin file is a modified version of the [solidity output json](https://solidity.readthedocs.io/en/v0.5.11/using-the-compiler.html#output-description).
+
+A solidity source file can have any number of contracts, and those contract names do not have to match the file name of the source. The resulting bin
+file(s) is named according to the name of the contract(s). To select which contracs to use, specifiy the _instance_ field.
```
Comment (gregdhill) [inline docs/reference/deploy-yaml.md]: typo

File: docs/reference/deploy-yaml.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+and will have to wait until the next block to retrieve the result; the query-contract job is for accessing read-only functions which do not require
+write access. This type of job has the following parameters:
+
+* _source_: the input address which should execute the transaction
+* _destination_: the account to access
+* _function_: the name of the function to call
+* _data_: the arguments to the function call
+* _bin_: the path to the abi or bin file
+
+The _destination_ field can either be a hex encoding adress, the name of a burrow keys key name (e.g. Root_0 or Participant_1), or the result
```
Comment (gregdhill) [inline docs/reference/deploy-yaml.md]: typo

File: docs/reference/deploy-yaml.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+of previous burrow deploy job. In the latter case the name of the job must be specified, prefixed with $.
+
+If the contract was deployed without metadata (e.g. using the burrow js module or with an earlier version of burrow deploy) the abi must be
+specified. This must be the path to the contract bin file or abi file.
+
+## proposal job
+
+This is described in the [proposal tutorial](../tutorials/8-proposals.md).
```
Comment (gregdhill) [inline docs/reference/deploy-yaml.md]: does this work correctly with docsify?

Comment (seanyoung) [reply docs/reference/deploy-yaml.md]: Yes, it does.

**Pull request merged**